nodes.tf

```
@@ -0,0 +1,15 @@
+module "nodegroups" {
+  for_each = var.nodegroups
+
+  source = "./modules/nodegroups"
+  node_count = each.value.count
+  name = "${var.name}-${each.value.role}"
```

I'd rather have the name set like the other modules:
```suggestion
  name = "${var.name}-${each.key}"
```